cpp/src/io/csv/reader_impl.cu

```diff
@@ -180,7 +183,19 @@ std::vector<std::string> setColumnNames(std::vector<char> const &header,
   return col_names;
 }
 
-table_with_metadata reader::impl::read(rmm::cuda_stream_view stream)
+template <typename C>
+void erase_except_last(C &container, rmm::cuda_stream_view stream)
```

Comment:
pulled this into a separate function because a device lambda cannot be declared in a private function